backend/modules/parsers/audioparser.py

```diff
@@ -0,0 +1,112 @@
+import json
+import os
+import tempfile
+from typing import Any, Dict, List
+
+from langchain.docstore.document import Document
+
+from backend.logger import logger
+from backend.modules.model_gateway.model_gateway import model_gateway
+from backend.modules.parsers.parser import BaseParser
+from backend.modules.parsers.unstructured_io import UnstructuredIoParser
+from backend.types import ModelConfig
+
+
+class AudioParser(BaseParser):
+    """
+    AudioParser is a parser class for extracting text from audio input.
+
+    {
+        ".mp3": {
+            "name": "AudioParser",
+            "parameters": {
+                "model_configuration": {
+                    "name" : "faster-whisper/Systran/faster-distil-whisper-large-v3"
+                },
+                "max_chunk_size": 2000
+            }
+        }
+    }
+
+    """
+
+    supported_file_extensions = [
+        ".flac",
+        ".mp3",
+        ".mp4",
+        ".mpeg",
+        ".mpga",
+        ".m4a",
+        ".ogg",
+        ".wav",
+        ".webm",
+    ]
+
+    def __init__(
+        self, *, model_configuration: ModelConfig, max_chunk_size: int = 2000, **kwargs
+    ):
+        """
+        Initializes the AudioParser object.
+        """
+        self.model_configuration = ModelConfig.model_validate(model_configuration)
+        self.audio_processing_svc = model_gateway.get_audio_model_from_model_config(
+            model_name=self.model_configuration.name
+        )
+        self.max_chunk_size = max_chunk_size
+        super().__init__(**kwargs)
+
+    async def get_chunks(
+        self, filepath: str, metadata: Dict[Any, Any] | None, **kwargs
+    ) -> List[Document]:
+        """
+        Get the chunks of the audio file.
+        """
+        try:
+            parsed_audio_text = []
+
+            response = self.audio_processing_svc.get_transcription(
+                audio_file_path=filepath
+            )
+
+            try:
+                for line in response.iter_lines():
+                    if line:
+                        data = json.loads(line.decode("utf-8").split("data: ")[1])[
+                            "text"
+                        ]
```

Comment:
Maybe we should just add sseclient-py as dependency for all SSE streaming
https://github.com/mpetazzoni/sseclient

Reply:
will consider this for future.